src/main/java/org/broadinstitute/dsm/util/PatchUtil.java

```diff
@@ -104,7 +106,14 @@ private void getColumnNames(Class<?> obj) {
                     if (StringUtils.isNotBlank(columnPrefix)) {
                         fieldKey = columnPrefix.concat("_").concat(field.getName());
                     }
-                    columnNameMap.put(fieldKey, new DBElement(tableName, tableAlias, primaryKey, column.value()));
+                    DbDateConversion dbDateConversion = field.getAnnotation(DbDateConversion.class);
+                    SqlDateConverter dateConverter = null;
+                    if (dbDateConversion != null) {
```

Comment:
in any case, does not matter `field.getAnnotation(DbDateConversion.class)` returned actual value or null, we pass dateConverter to `DBElement` constructor, what if we can eliminate if check for null, something like that: 
`SqlDateConverter dateConverter = field.getAnnotation(DbDateConversion.class).value();`

Reply:
Won't this throw an NPE if there's no annotation for the field?

Reply:
We could pass `DbDateConversion` to the `DBElement` constructor and do the null check there--that would make calling code a bit cleaner.

Reply:
oh good catch, my bad sorry, about null checks usually I prefer to use `Optional` getters for the fields, something like:
```
     public class DBElement {
       private SqlDateConverted dateConverter;
       
       public Optional<SqlDateConverted> getDateConverter() {
            return Optional.ofNullable(this.dateConverter);
       }
     }
```
and caller will have to handle it with methods of `Optional` which provides much cleaner code from my point of view.